### PR TITLE
Remove all createClass instances using codemods take 2

### PR DESCRIPTION
**Pull request by samouri**
This PR seeks to remove all usages of the deprecated `createClass` from the reader codebase.
It does so by taking advantage of the `class` codeshift defined in [react-codemods](https://github.com/reactjs/react-codemod).

Notes:
- [wp-prettier](https://github.com/samouri/prettier) is a fork of prettier that keeps all of calypso's spacing. it means that there aren't any extraneous whitespace diffs in this PR
- I initially did it differently (see [take1](https://github.com/Automattic/wp-calypso/pull/13948)) but then learned it could all be done with a single smarter codemod that would even transform to PureComponent
- there are 4 createClass usages that couldn't be fixed automatically. I've added the `react-create-class` compat package as a dependency and we can remove those 4 in future PRs.

```
jscodeshift -t ../react-codemod/transforms/class.js --mixin-module-name=react-pure-render/mixin --remove-runtime-proptypes=false --extensions=js,jsx --pure-component=true client/reader
git commit -nam "run class codemod "
./../prettier/bin/prettier.js  --write client/reader/**.js* 
./../prettier/bin/prettier.js  --write client/reader/**/*.js* 
git commit -nam "run wp-prettier" 
npm i --save create-react-class 
make shrinkwrap
```

### CODE REVIEW

File: client/reader/recommendations/for-you/index.jsx [hunk 1]
```diff
@@ -21,61 +21,63 @@ import { recordAction, recordGaEvent, recordTrack } from 'reader/stats';
 import { getSiteUrl } from 'reader/route';
 import { decodeEntities } from 'lib/formatting';
 
-const RecommendedForYou = React.createClass( {
-	getInitialState() {
+class RecommendedForYou extends React.Component {
+	constructor( props, context ) {
```
Comment (bluefuton) [inline client/reader/recommendations/for-you/index.jsx]: Not sure it needs `context`, but also not doing any harm.

Comment (samouri) [inline client/reader/recommendations/for-you/index.jsx]: agreed.

**Comment by gziolo** [comment]
Nice 🥇

**Comment by gziolo** [comment]
Pinging @ockham @jsnajdr @jsnmoon to let them know how prettier plays nicely with codemods :)

**Comment by jsnajdr** [comment]
Congrats! One question about running the `calypso-prettier`: do I need to specify the `--print-width 140` and `--tab-width 4` options? Or does the fork change the defaults from 80 and 2, respectively?

**Comment by samouri** [comment]
@jsnajdr, the fork ignores absolutely everything you tell it.  You can run it without any args.  I've forced 100 print-width and 2 tab-width.

**Pull request merged**